Makefile

```diff
@@ -0,0 +1,39 @@
+export VENV_PATH := .venv
+# Use python executables inside venv
+export PATH := $(VENV_PATH)/bin:$(PATH)
+export PYTHONPATH=.
+
+# creates the venv
+$(VENV_PATH)/bin/python3.11:
+	python3.11 -m venv $(VENV_PATH)
+
+# makes sures the venv contains a given version of pip and pip-tools
+$(VENV_PATH): $(VENV_PATH)/bin/python3.11
+	pip install --quiet --upgrade 'pip==23.3.1' 'pip-tools==7.3'
+
+# generates a lock file with pinned version of all dependencies to be used by the CI and local devs
+requirements/dev.txt: $(VENV_PATH) requirements/dev.in pyproject.toml
+	pip-compile \
+		--quiet --generate-hashes --max-rounds=20 --strip-extras \
+		--resolver=backtracking \
+		--output-file requirements/dev.txt \
+		 requirements/dev.in pyproject.toml
+
+# upgrades the dependencies to their latest/matching version
+upgrade: $(VENV_PATH)
```

Comment:
Should this be `.PHONY` too? Judging from `install`

Reply:
yes, indeed, [phony targets ](https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html) dhall be used for names that are not files